controllers/podplacement/pod_model.go

```diff
@@ -172,13 +172,14 @@ func (pod *Pod) setRequiredArchNodeAffinity(requirement corev1.NodeSelectorRequi
 }
 
 // SetPreferredArchNodeAffinity sets the node affinity for the pod to the preferences given in the ClusterPodPlacementConfig.
-// TODO[Tori]: Missing unit tests for this method.
 func (pod *Pod) SetPreferredArchNodeAffinity(cppc *v1beta1.ClusterPodPlacementConfig) {
 	// Prevent overriding of user-provided kubernetes.io/arch preferred affinities
 	if pod.isPreferredAffinityConfiguredForArchitecture() {
-		pod.ensureLabel(utils.PreferredNodeAffinityLabel, utils.LabelValueNotSet)
-		pod.publishEvent(corev1.EventTypeNormal, ArchitectureAwareNodeAffinitySet,
-			ArchitecturePreferredPredicateSkippedMsg)
+		if _, exists := pod.Labels[utils.PreferredNodeAffinityLabel]; !exists {
```

Comment:
```suggestion
		if v, exists := pod.Labels[utils.PreferredNodeAffinityLabel]; !exists || v == utils.LabelValueSet {
```

do we maybe need to be more specific?